packages/@react-native-windows/telemetry/src/utils/errorUtils.ts

```diff
@@ -0,0 +1,157 @@
+  // SDK requirements
+  MinSDKVersionNotMet: 2200,
+  BadSDKVersionFormat: 2201,
+  NoSDK: 2202,
+  // Build
+  NoMSBuild: 2300,
+  NoVSWhere: 2301,
+  MSBuildError: 2302,
+  // Deploy
+  NoAppPackage: 2400,
+  NoAppxManifest: 2401,
+  NoDevice: 2402,
+  AppDidNotDeploy: 2403,
+  InvalidDevicesOutput: 2404,
+  RemoveOldAppVersionFailure: 2405,
+  EnableDevModeFailure: 2406,
+  InstallAppFailure: 2407,
+  InstallAppDependenciesFailure: 2408,
+  CheckNetIsolationFailure: 2409,
+  InstallAppToDeviceFailure: 2410,
+  UninstallAppOnDeviceFailure: 2411,
+  DeployRecipeFailure: 2412,
+  // Launch
+  AppStartupFailure: 2500,
+
+  // autolink-windows
+  NoWindowsConfig: 3000,
+  IncompleteConfig: 3001,
+  InvalidConfig: 3002,
+  NeedAutolinking: 3003,
+  AddProjectToSolution: 3004,
+  Autolinking: 3005,
+};
+
+export type CodedErrorType = keyof typeof CodedErrors;
+
+/**
+ * Represents an error whose message might contain user-originating content,
+ * therefore when transmitting telemetry, only the type should be sent.
+ * @param type a stable ID identifying the type of error.
+ * @param message the error text. This should only be used for display to the user.
+ * @param data any additional metadata that is safe to collect for telemetry purposes.
+ */
+export class CodedError extends Error {
+  constructor(
+    public readonly type: CodedErrorType,
+    message: string,
+    public readonly data?: Record<string, any>,
+  ) {
+    super(message);
+    this.name = type;
+  }
+}
+
+/**
+ * Tries to parse an error code out of an error message.
+ * @param msg An error message to process.
+ * @returns The parsed error code.
+ */
+export function tryGetErrorCode(msg: string): string | undefined {
+  const errorRegEx = /error (\w+\d+):/gi;
+  const m = errorRegEx.exec(msg);
+  return m ? m[1] : undefined;
+}
+
+/**
+ * Sanitize an error message by anonymizing any paths that appear between quotes (''), brackets ([]), or double quotes ("").
+ * @param msg The error message to sanitize.
+ * @return The message with any paths anonymized.
+ */
+export function sanitizeErrorMessage(msg: string): string {
+  const cpuThreadId = /^\d+(:\d+)?>/g;
+  msg = msg.replace(cpuThreadId, '');
+  const parts = msg.split(/['[\]"]/g);
+  const clean = [];
+  const pathRegEx = /(['["]?)([A-Za-z]:|\\)[\\/]([^<>:;,?"*\t\r\n|/\\]+[\\/])+([^<>:;,?"*\t\r\n|]+\/?(['["]?))/gi;
+  for (const part of parts) {
+    if (pathRegEx.test(part)) {
+      pathRegEx.lastIndex = -1;
+      let matches: RegExpExecArray | null;
+      let noPath = '';
+      let last = 0;
+      while ((matches = pathRegEx.exec(part))) {
+        noPath +=
+          part.substr(last, matches!.index - last) +
+          sanitizeUtils.getAnonymizedPath(matches[0]);
+        last = matches!.index + matches![0].length;
+      }
+      if (noPath !== '') {
+        clean.push(noPath);
+      }
+    } else if (part !== '') {
+      clean.push(part);
+    }
+  }
+  return clean.join('').trim();
+}
```

Comment:
This function is not safe to be used, since it by design only tries to remove some recognized patterns (and misses some important ones). We only enable it with `preserveErrorMessages`, but if full messages are opt-in, should we still try to scrub them?

Reply:
This just sanitizes file paths and we have lots of tests for that. Do we need to do this when our current default is to strip all messages? No. This code won't even get run. Will we need this if we flip the switch, yes. If there are other patterns let's add them before we ever decide we need to flip the switch.